binapigen/generator.go

```diff
@@ -34,17 +34,42 @@ import (
 	"go.fd.io/govpp/binapigen/vppapi"
 )
 
+var Logger = logrus.New()
+
+func init() {
+	if debug := os.Getenv("DEBUG_GOVPP"); strings.Contains(debug, "binapigen") {
+		Logger.SetLevel(logrus.DebugLevel)
+		logrus.SetLevel(logrus.DebugLevel)
+	}
+}
+
+func logf(f string, v ...interface{}) {
+	Logger.Debugf(f, v...)
+}
+
+func (*Generator) Logf(f string, v ...interface{}) {
+	logf(f, v...)
+}
+
+type Options struct {
+	OutputDir         string // output directory for generated files
+	ImportPrefix      string // prefix for import paths
+	NoVersionInfo     bool   // disables generating version info
+	NoSourcePathInfo  bool   // disables the 'source: /path' comment
+	ActivePluginNames []string
+	ApiDir            string
```

Comment:
Directory with API files should not really be an input parameter for the Generator. It should work only with the `[]vppapi.File` and let caller load & parse from directory. This makes the `Generator` very simple, the input parameters are not coupled with filesystem and expect only a pre-defined data format. This way we can use Generator to work in runtime, e.g. load API files from specific commit of some git repo, generate the JSON API files, load them and use as input to Generator.. or retrieve the API schemas directly from VPP when running without having any generated Go bindings before hand - the ultimate client for VPP.